bql/grammar/grammar.go

```diff
@@ -365,6 +366,33 @@ func clauses() []*Clause {
 	}
 }
 
+func moreFilterClauses() []*Clause {
```

Comment:
it looks like "more filter clauses" is part of the business logic (grammar in this case). I'm wondering if it make sense to be more specific as "more filters" is too generic. A common question would be more than what?. How much more filters? What's the difference from these "more filters" and the normal "filters".

Reply:
I see what you mean. The BadWolf grammar is a left LL(1) factorized grammar as commented in the beginning of the `grammar.go` file, which means that it is defined recursively looking one step ahead each time during the parsing. 

This way, the `moreFilterClauses` comes just to recursively open a path to more filter clauses in the grammar, during parsing, as it is common in this type of grammar and as it is already done for other functions in this file (such as `moreClauses` and `moreGraphsClauses`).

Since this is intrinsically related to the way this kind of recursive grammar works, and we already have other examples like this in the file, I am inclined to let it as it is. But ltm if you still think we should make any other changes here. : )

Reply:
LGTM